File: student_frontend/lib/model/transaction_dto.dart
import 'package:student_frontend/model/transaction_details.dart';

class TransactionDTO {
  final int transactionId;
  final int payerWalletId;
  final int receiverWalletId;
  final String createdTime;
  final String transactionStatus;
  final double totalPrice;
  final List<TransactionDetail> transactionDetails;

  TransactionDTO({
    required this.transactionId,
    required this.payerWalletId,
    required this.receiverWalletId,
    required this.createdTime,
    required this.transactionStatus,
    required this.totalPrice,
    required this.transactionDetails,
  });

  factory TransactionDTO.fromJson(Map<String, dynamic> json) {
    return TransactionDTO(
      transactionId: json['transactionId'],
      payerWalletId: json['payerWalletId'],
      receiverWalletId: json['receiverWalletId'],
      createdTime: json['createdTime'],
      transactionStatus: json['transactionStatus'],
      totalPrice: json['totalPrice'].toDouble(),
      transactionDetails: List<TransactionDetail>.from(
          json['transactionDetails'].map((x) => TransactionDetail.fromJson(x))),
    );
  }
}

File: backend/Backend/src/main/java/com/sgw/backend/dto/TransactionDTO.java
package com.sgw.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import com.sgw.backend.entity.TransactionStatusEnum;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TransactionDTO {

    private long transactionId;
    private Long payerWalletId;
    private Long receiverWalletId;
    private LocalDateTime createdTime;
    private TransactionStatusEnum transactionStatus;
    private BigDecimal totalPrice;
    private List<TransactionDetailDTO> transactionDetails;
}
